scripts/listener.ts

```diff
@@ -119,68 +175,192 @@ async function getTokenLists() {
-    MarlinEvents.subscribeEvents({
+  } else if (chainSupportedBy(network, MarlinEvents.Types.EventChains)) {
+    const contracts = {
+      comp: '0xEa2923b099b4B588FdFAD47201d747e3b9599A5f', // TESTNET
+      governorAlpha: '0xeDAA76873524f6A203De2Fa792AD97E459Fca6Ff', // TESTNET
+      timelock: '0x7d89D52c464051FcCbe35918cf966e2135a17c43', // TESTNET
+    };
+    const api = await MarlinEvents.createApi(listenerArg.url, contracts);
+    return MarlinEvents.subscribeEvents({
       chain: network,
       api,
-      handlers: [new StandaloneEventHandler()],
-      skipCatchup,
+      handlers: handlers,
+      skipCatchup: listenerArg.skipCatchup,
       verbose: true,
     });
-  });
-} else if (chainSupportedBy(network, Erc20Events.Types.EventChains)) {
-  async function erc20Subscribe() {
+  } else if (chainSupportedBy(network, Erc20Events.Types.EventChains)) {
     let tokens = await getTokenLists();
     let tokenAddresses = tokens.map((o) => o.address);
-    Erc20Events.createApi(url, tokenAddresses).then((api) => {
-      Erc20Events.subscribeEvents({
-        chain: network,
-        api,
-        handlers: [new StandaloneEventHandler()],
-        skipCatchup,
-        verbose: true,
-      });
+    const api = await Erc20Events.createApi(listenerArg.url, tokenAddresses);
+    return Erc20Events.subscribeEvents({
+      chain: network,
+      api,
+      handlers: handlers,
+      skipCatchup: listenerArg.skipCatchup,
+      verbose: true,
     });
   }
-  erc20Subscribe();
-}
\ No newline at end of file
+}
+
+/**
+ * Creates a listener instance for any chain. NOTE: this function has 2 main side effects.
+ * 1. It processes the options parameter to populate the listenerArgs object.
+ * 2. It saves the subscriber returned by setupListener in the subscribers object.
+ * This function is useful to start a listener from scratch when given unprocessed options
+ * @param chain The name of the chain to listen to
+ * @param options An object containing the desired settings for the listener (see ReadMe.md)
+ */
+export async function createListener(chain, options: any) {
+  listenerArgs[chain] = {
+    archival: !!options.archival,
+    startBlock: options.startBlock ?? 0,
+    url: options.url || networkUrls[chain],
+    spec: networkSpecs[chain] || {},
+    contract: options.contractAddress || contracts[chain],
+    skipCatchup: !!options.skipCatchup,
+    excludedEvents: options.excludedEvents || [],
+  };
+
+  if (SubstrateChains.includes(chain)) {
+    try {
+      const newSpec = await getSubstrateSpecs(chain as EventSupportingChainT);
+      if (newSpec?.types != undefined) listenerArgs[chain].spec = newSpec;
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  try {
+    subscribers[chain] = await setupListener(
+      chain as EventSupportingChainT,
+      listenerArgs[chain]
+    );
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+export let listenerArgs: { [key: string]: IListenerOptions } = {};
```

Comment:
These should be consts fwiw, since you can modify the contents of the object without reassigning it.